packages/lisk-p2p/src/p2p.ts

```diff
@@ -480,6 +497,52 @@ export class P2P extends EventEmitter {
 		return seedPeerUpdatedInfos;
 	}
 
+	private _pickRandomDiscoveredPeers(
+		count: number,
+	): ReadonlyArray<P2PDiscoveredPeerInfo> {
+		const discoveredPeerList: ReadonlyArray<P2PDiscoveredPeerInfo> = [
+			...this._newPeers.values(),
+			...this._triedPeers.values(),
+		]; // Peers whose values has been updated atleast once.
+
+		return selectRandomPeerSample(discoveredPeerList, count);
+	}
+
+	private _handleGetAllPeersRequest(request: P2PRequest): void {
+		// TODO later: Remove fields that are specific to the current Lisk protocol.
+		const protocolPeerInfoList: ProtocolPeerInfoList = {
+			success: true,
+			peers: this._pickRandomDiscoveredPeers(MAX_PEER_LIST_BATCH_SIZE)
+				.map(
+					(peerInfo: P2PDiscoveredPeerInfo): ProtocolPeerInfo | undefined => {
```

Comment:
would it be possible to have `undefined` here? seems like `_pickRandomDiscoveredPeers` must return array